spec.md

```diff
@@ -230,6 +245,39 @@ A common use of an externally defined module is to represent a Verilog module
 that will be written separately and provided together with FIRRTL-generated
 Verilog to downstream tools.
 
+## Public Modules
+
+A public module is accessible by `Circuit`s other than the `Circuit` it is
+defined in. Unless otherwise specified, all modules inside of a `Circuit` are
+private and cannot be accessed by any other `Circuit`.
+
+A public module has fixed ports.
+
+A module can be marked as public like so:
+
+```firrtl
+circuit FooCircuit :
+   module Foo is public :
```

Comment:
Not sure `is public` is what we want here (that was just a sketch from me), probably just `public module` is more "what people are used to"?